### ISSUE TITLE
Windows syscheck scan never ends if realtime folder doesn't exists

### ISSUE THREAD

**Issue by migruiz4** [wazuh/wazuh#2991]
|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| 3.8.2-1 | Agent | Packages | Windows |

Hi team,

Syscheck scan gets stuck if a non-existing folder is configured with real-time in Windows agents.

These are the steps to reproduce the issue:

Create a group and add a Windows agent (in my case Windows Server 2012-R2)
Add the following configuration:
```xml
<agent_config>

    <!-- Shared agent configuration here -->
    <syscheck>
      <directories check_all="yes" realtime="yes">C:\FIM TUNING</directories>
      <directories check_all="yes" realtime="yes">C:\dontlookhere</directories>
    </syscheck>

</agent_config>
```

This causes the syscheck scan to never end, not reporting anything and not starting the realtime monitoring.

**Comment by DaveVG1** [comment]
Hi Miguel,

This has been fixed in 3.9 (https://github.com/wazuh/wazuh/commit/2841e549cbf3b79021b9a1c0c7672511f5c39a62).

A mutex wasn't being unlocked, causing syscheck to never end.

Regards.

**Comment by migruiz4** [comment]
Hi David,

That is great news. 

Closing the issue then.
